var/spack/repos/builtin/packages/glew/package.py

```diff
@@ -18,18 +19,41 @@ class Glew(CMakePackage):
     version('2.1.0',  sha256='04de91e7e6763039bc11940095cd9c7f880baba82196a7765f727ac05a993c95')
     version('2.0.0',  sha256='c572c30a4e64689c342ba1624130ac98936d7af90c3103f9ce12b8a0c5736764')
 
+    variant('gl',
+            default='glx' if sys.platform.startswith('linux') else 'other',
+            values=('glx', 'osmesa', 'other'), multi=False,
+            description='The OpenGL provider to use')
+    conflicts('osmesa', when='gl=glx')
+    conflicts('osmesa', when='gl=other')
+    conflicts('glx', when='gl=osmesa')
+    conflicts('glx', when='gl=other')
```

Comment:
I'm a bit confused by these conflicts directives. They seem to say that this package conflicts with being named "osmesa", which will never happen. Are they supposed to say `'^osmesa'` meaning conflict with depending on osmesa or `'gl=osmesa'`?

Reply:
The variant `gl=<X>` is a variant for this package, it is not something special that spack interprets at the same level as `osmesa` or `^osmesa`. The point of the conflicts is to translate that variant correctly to the virtual packages so that an invalid combination does not get built.

Reply:
But osmesa and glx are concrete packages, not virtual ones, and glew does not provide either of them, does it? If I'm interpreting the conflict checking correctly, saying `conflicts('osmesa', ...)` translates into checking whether the concretized spec for glew `glewSpec.satisfies('osmesa')` and I don't see how that can ever be the case.

Reply:
The conflict is just saying you cannot build a `glew gl=glx` if osmesa is also in the stack, because that isn't compatible. And that can happen because mesa can provide both glx and osmesa backends in the same build. You try removing these and building glew with osmesa and paraview with ~osmesa you can get a build with mixed backends which is not good. These conflicts prevent those cases.